### PR TITLE
Use of insecure HostKeyCallback implementation

### PR DESCRIPTION
**Pull request by riteshsonawane1372**
# Tasks

- #47 

Use of insecure HostKeyCallback implementation

# Solution :heavy_check_mark:
```go
HostKeyCallback: ssh.HostKeyCallback(
	func(hostname string, remote net.Addr, key ssh.PublicKey) error {
		return nil
}),
```

# Screenshots :framed_picture:

# Note to reviewers :notebook:

### CODE REVIEW

**Comment by dipankardas011** [comment]
@riteshsonawane1372 are there no other methods available?

**Comment by riteshsonawane1372** [comment]
`` FixedHostKey `` : When the allow list contains only a single host key

`` HostKeyCallback: ssh.FixedHostKey(publicKey),``

**Comment by riteshsonawane1372** [comment]
```
func secureHostKeyCallback() {
	publicKeyBytes, _ := ioutil.ReadFile("allowed_hostkey.pub")
	publicKey, _ := ssh.ParsePublicKey(publicKeyBytes)

	_ = &ssh.ClientConfig{
		User:            "username",
		Auth:            []ssh.AuthMethod{nil},
		HostKeyCallback: ssh.FixedHostKey(publicKey),
	}
}

```

**Comment by dipankardas011** [comment]
@riteshsonawane1372 could you replace the password-based auth with ssh key auth?
if you are up for it

File: api/civo/utils.go [hunk 1]
```diff
@@ -268,7 +272,10 @@ func ExecWithOutput(publicIP, password, script string, fastMode bool) (string, e
 		},
 		//HostKeyCallback: hostKeyCallback,
 		// FIXME: Insecure Ignore should be replaced with secure
```
Comment (dipankardas011) [inline api/civo/utils.go:274]: Remove the comment

**Comment by riteshsonawane1372** [comment]
> @riteshsonawane1372 could you replace the password-based auth with ssh key auth? if you are up for it

``` 
	signer, _ := ssh.ParsePrivateKey([]byte(password))
	config := &ssh.ClientConfig{
		User: "root",
		Auth: []ssh.AuthMethod{
			ssh.PublicKeys(signer),
		},
		HostKeyCallback: ssh.HostKeyCallback(
			func(hostname string, remote net.Addr, key ssh.PublicKey) error {
				return nil
		

```

**Comment by dipankardas011** [comment]
> > @riteshsonawane1372 could you replace the password-based auth with ssh key auth? if you are up for it
> 
> ``` 
> 	signer, _ := ssh.ParsePrivateKey([]byte(password))
> 	config := &ssh.ClientConfig{
> 		User: "root",
> 		Auth: []ssh.AuthMethod{
> 			ssh.PublicKeys(signer),
> 		},
> 		HostKeyCallback: ssh.HostKeyCallback(
> 			func(hostname string, remote net.Addr, key ssh.PublicKey) error {
> 				return nil
> 		
> 
> ```

If we can move to private key Auth
We have to create sshkeypair function and delete when delete cluster
Get the key and save it to ksctl cluster folder and use it to ssh
Ig that will match with rest of the cloud providers

**Comment by dipankardas011** [comment]
- Issue #61 
@AvineshTripathi

**Comment by riteshsonawane1372** [comment]
> > > @riteshsonawane1372 could you replace the password-based auth with ssh key auth? if you are up for it
> > 
> > 
> > ```
> > 	signer, _ := ssh.ParsePrivateKey([]byte(password))
> > 	config := &ssh.ClientConfig{
> > 		User: "root",
> > 		Auth: []ssh.AuthMethod{
> > 			ssh.PublicKeys(signer),
> > 		},
> > 		HostKeyCallback: ssh.HostKeyCallback(
> > 			func(hostname string, remote net.Addr, key ssh.PublicKey) error {
> > 				return nil
> > 		
> > ```
> 
> If we can move to private key Auth We have to create sshkeypair function and delete when delete cluster Get the key and save it to ksctl cluster folder and use it to ssh Ig that will match with rest of the cloud providers @AvineshTripathi @kranurag7 your thoughts

Can you describe a bit ?

**Comment by dipankardas011** [comment]
> > > > @riteshsonawane1372 could you replace the password-based auth with ssh key auth? if you are up for it
> > > 
> > > 
> > > ```
> > > 	signer, _ := ssh.ParsePrivateKey([]byte(password))
> > > 	config := &ssh.ClientConfig{
> > > 		User: "root",
> > > 		Auth: []ssh.AuthMethod{
> > > 			ssh.PublicKeys(signer),
> > > 		},
> > > 		HostKeyCallback: ssh.HostKeyCallback(
> > > 			func(hostname string, remote net.Addr, key ssh.PublicKey) error {
> > > 				return nil
> > > 		
> > > ```
> > 
> > 
> > If we can move to private key Auth We have to create sshkeypair function and delete when delete cluster Get the key and save it to ksctl cluster folder and use it to ssh Ig that will match with rest of the cloud providers @AvineshTripathi @kranurag7 your thoughts
> 
> Can you describe a bit ?

ig it will be clear if you can check the Issue created still any doubt we can discuss

**Comment by dipankardas011** [comment]
@riteshsonawane1372 you are working on main branch do checkout to a different branch
I am closing this PR you can create a another PR as this PR will break the main branch code
ig some miscommunication between us
we can communicate what needs to be done (i.e. the workflow and how to change the src code)

**Pull request closed without merging**